src/concepts/plugins/frontend.md

```diff
@@ -0,0 +1,479 @@
+# Plugin Frontend
+
+The frontend client component of Caido is the application running on your device.
+
+Frontend development allows you to:
+
+- Enhance the user-interface/experience.
+- Add new pages, components and elements.
+- Modify the appearance, behavior and functionality of the user-interface.
+- Provide additional features and customization options.
+- Handle user interactions, render data and communicate with the backend server via Caido's API.
+
+## Frontend Interfaces
+
+_For advanced documentation on this topic - click [here](./frontend_sdk.md)._
+
+`ui` - Used to create **cards** (_bordered boxes with padding around their content_), **wells** (_a rounded border around an element with a gray background color and some padding_) and buttons. Content options for each element are also provided. These elements provide a visually appealing way to sectionalize the user-interface of your plugin.
+
+`scopes` - Used to set, create, update and delete **target scoping rules**, ensuring your plugin is directed at desired hosts.
+
+`commands` - Used to register **actions** to expose functionality, bind actions to the user-interface and implement business logic.
+
+`menu` - Used to register right-click **context menu** actions/options, allowing quick access to your plugin functionality.
+
+`navigation` - Used to create pages in the application, giving your plugin its own **tab**.
+
+`window` - Used to interact with **text** within the application environment, allowing highlight to action flows.
+
+`storage` - Used to **persist data** across different sessions or instances.
+
+::: info
+Caido has heavily referenced [Visual Studio Code's Command Model](https://code.visualstudio.com/api/extension-guides/command).
+:::
+
+::: tip
+Be aware that your plugin can be loaded in multiple user tabs.
+:::
+
+## Frontend Starterkit Repository Contents
+
+::: info
+
+- For documentation on the tooling files shared by all plugin starterkits offered by Caido - click [here](/concepts/plugins/plugin_tooling.md).
+- The frontend starterkit can be found [here](https://github.com/caido/starterkit-plugin-frontend).
+:::
+
+### Frontend Starterkit Directories
+
+The `public` directory stores the `styles.css` file used to stylize elements of your plugin.
+
+The `src` directory stores the following files:
+
+- `index.ts`: This file acts as the entrypoint file (the initial script that is loaded and executed, setting up the necessary resources and handling further logic and interactions). View the script breakdown below.
+- `types.ts`: This file defines and exports type delcarations/definitions to be bundled with your code and be consumable by dependents.
+
+::: info
+You may only have one CSS and entrypoint file per plugin package.
+:::
```

Comment:
That is true but it might be confusing, so I would add `after the build process` or maybe even move it in the manifest section.